sycl/doc/extensions/DiscardQueueEvents/SYCL_EXT_ONEAPI_DISCARD_QUEUE_EVENTS.asciidoc

```diff
@@ -0,0 +1,223 @@
+Performance improvement is the main motivation for using this property.
+
+Below is a usage example:
+[source,c++]
+----
+  sycl::property_list props{ext::oneapi::property::queue::discard_queue_events{}
+                            property::queue::in_order{}};
+  sycl::queue Queue( props );
+
+  // some USM preparations ..
+
+  sycl::event e1, e2, e3;
+
+  // returning "invalid" events from each submission function:
+  e1 = Queue.parallel_for(NDRange, [=](nd_item<1> item){ do_smth1(); });
+
+  e2 = Queue.single_task([=](){ do_smth2(); });
+
+  e3 = Queue.submit([&](handler &CGH) { CGH.parallel_for(NDRange, [=](nd_item<1> item){ do_smth3(); }); });
+
+  Queue.wait();
+----
+
+In the example above, the application doesn't use sycl events: `e1`, `e2`, `e3`
+and is waiting for the end of work by `queue::wait()`. The returned events will be
+_invalid_ events. _invalid_ event is a `sycl::event` with a status `invalid` and the event returns from each kernel
+submission when queue is created with the `discard_queue_events` property.
+See the description of behavior for this event below for details.
+
+_invalid_ event API behavior:
+[source,c++]
+----
+// the behavior is the same as for the default event, i.e. must throw an exception with the errc::invalid_object error code
+cl_event get() const
+
+// the behavior is the same as for the default event, i.e. must return true
+bool is_host() const;
+
+// the behavior is the same as for the default event, i.e. must return backend::host
+backend get_backend() const noexcept;
+
+// the behavior is the same as for the default event, i.e. must return a zero-sized vector
+std::vector<event> get_wait_list();
+
+// must throw an exception with the errc::invalid_object error code.
+void wait();
+
+// passing invalid event in the function must throw an exception with the errc::invalid_object error code.
+static void wait(const std::vector<event> &eventList);
+
+// must throw an exception with the errc::invalid_object error code.
+void wait_and_throw();
+
+// passing invalid event in the function must throw an exception with the errc::invalid_object error code.
+static void wait_and_throw(const std::vector<event> &eventList);
+
+// returns info::event_command_status::ext_oneapi_unknown status
+get_info<info::event::command_execution_status>() const;
+
+// the behavior is the same as for the default event, i.e. returns 0
+get_info<info::event::reference_count>() const;
+
+// must throw an exception with the errc::invalid_object error code.
+template <info::event_profiling param>
+typename param::return_type get_profiling_info() const;
+----
+
+The behavior when passing _invalid_ event to handler API:
+[source,c++]
+----
+// must throw an exception with the errc::invalid_object error code.
+handler::depends_on(event Event)
+
+// must throw an exception with the errc::invalid_object error code.
+handler::depends_on(const std::vector<event> &Events)
+----
+
+== Limitations

```

Comment:
```suggestion
== DPCPP implementation limitations
```
This is planned limitations for the first version of implementation which we may lift in future, right?
If so, maybe rename it to something like this?

Reply:
Right, I will update.